interfaces/apparmor/backend.go

```diff
@@ -216,7 +217,11 @@ func (b *Backend) deriveContent(spec *Specification, snapInfo *snap.Info, opts i
 
 func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.ConfinementOptions, snippetForTag string, content map[string]*osutil.FileState) {
 	var policy string
-	if opts.Classic && !opts.JailMode {
+	// When apparmor partial then use the classic template as we cannot enable
+	// effective confinement and we cannot rely on the devmode template which
+	// may have things that local apparmor doesn't understand.
```

Comment:
This comment isn't quite right. Note that classicTemplate has bare rules for all the modern rules so an old parser will always choke on them. This means that partial confinement must have new tools, and those new tools know how to work with all kernels. As such, this should read as:

```
// When partial AppArmor is detected, use the classic template for now. We could
// use devmode, but that could generate confusing log entries for users running
// snaps on systems with partial AppArmor support.
```

Reply:
+1, thank you.